otelcollector/test/utils/kubernetes_api_utils.go

```diff
@@ -306,6 +306,62 @@ func CheckIfAllContainersAreRunning(clientset *kubernetes.Clientset, namespace,
 	return nil
 }
 
+/*
+ * Check that pods are scheduled in all the nodes. If a node has no schduled pod on it, return an error.
+ */
+func CheckIfAllPodsScheduleOnNodes(clientset *kubernetes.Clientset, namespace, labelKey string, labelValue string) error {
+
+	// Get list of all nodes
+	nodes, err := clientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+
+	if err != nil {
+		return errors.New(fmt.Sprintf("Error getting nodes with the specified labels: %v", err))
+	}
+
+	for _, node := range nodes.Items {
+
+		// Get list of pods scheduled on this node
+		pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
+			FieldSelector: "spec.nodeName=" + node.Name,
+		})
+
+		if err != nil || pods == nil || len(pods.Items) == 0 {
```

Comment:
This checks if any pods are on the node, but we would want to make sure the ama-metrics daemonset pod exists on this node with the labelkey and labelvalue "dsName" = "ama-metrics-node"

Reply:
Added the label selector as a an additional option to get the pods. Thanks!